docs/DeveloperGuide.md

```diff
@@ -2,28 +2,454 @@
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the
+original source as well}
 
-## Design & implementation
+- [AB3](https://github.com/se-edu/addressbook-level3)
+- [OpenCSV](https://opencsv.sourceforge.net/)
+- [gson](https://github.com/google/gson)
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## Introduction 
 
+Welcome to rainyDay's developer guide! rainyDay is a desktop application for managing your finances that runs on a Command Line Interface (CLI). rainyDay
+provides a simple solution to track your finances and achieve your financial goals. 
+
+#### Main functions:
+* Adding, deleting and editing transaction entries
+* Viewing your transactions and filtering by certain fields
+* Saving your data into a csv file
+* Setting budgeting goals
+
+#### Purpose and target reader
+
+This guide is designed for potential engineers  and professors who may want to who may want to understand how rainyDay is implemented. 
+We assume that you already have a certain level of technical knowledge as we will be going into detail of the technical aspects
+of rainyDay. <br> If you are a user looking for instructions on how to use rainyDay, please refer to our [UserGuide](https://ay2223s2-cs2113t-t09-1.github.io/tp/UserGuide.html)
+
+#### How to use this guide
+
+{todo, talk about the diagrams, formatting (markdown etc)}
+
+## Design
+
+> 💡**Tip:** The .puml files used to create diagrams in this document can be found [here](https://github.com/AY2223S2-CS2113T-T09-1/tp/tree/master/docs/diagrams). 
+> Refer to the PlantUML Tutorial at se-edu/guides to learn how to create and edit diagrams.
+
+### Architecture
+
+#### Main components of the architecture
+- `Main`: Initializes the components in the correct sequence, and connects them up with each other
+- `Parser`: breaks down inputs passed from user
+- `Command`: executes the appropriate commands
+- `Userdata`: Data input by the user
+- `UI`: the user interface
+- `Storage`: reads data from and writes data to the hard disk file
+
+The diagram given below explains the high-level design of rainyDay, and how the components are related to one another.
+
+![architecture.png](images%2Farchitecture.png)
+
+The Sequence Diagram below shows the timeline of how the different components will interact with one another, we will
+illustrate a 'delete' command, e.g. `delete 1`.
+
+![ArchitectureSequence.png](images%2FArchitectureSequence.png)
```

Comment:
The sequence diagram is clean, but I'm not sure what this section of the sequence diagram is supposed to represent:

![image](https://user-images.githubusercontent.com/69704055/228727347-2a0c8c79-4db5-4186-bf58-86389e5cf313.png)

There does not seem to be a method call associated to this line and I am not sure what information this part is trying to convey. Could this be trying to represent a [constructor](https://nus-cs2113-ay2223s2.github.io/website/se-book-adapted/chapters/uml.html#object-creation)?

Reply:
Additionally, to maintain consistency in the sequence diagram, might it be better to have all solid arrows show a method?  

For example, in this part of the sequence diagram, the method name is correctly named (with the `()`):
![image](https://user-images.githubusercontent.com/69704055/228728346-6cabe9d4-f0fe-4cf4-8a76-361d9dd1033d.png)



However, this callback seems to be not following the proper syntax:
![image](https://user-images.githubusercontent.com/69704055/228727969-c80e9a13-9d98-4e62-b53e-551120fd2b65.png)